### PR TITLE
Automate publishing to NPM

### LINKED ISSUE #118: Automate package publishing to NPM
Publishing the packages to NPM involves [a lot of subtle steps](https://www.notion.so/liveblocks/How-to-074d3dfa5b944adb986fe176572d6887#9a10b3d58f014982a513f667ca5d67bd) that are currently carried out manually. There's a lot of room for error if you miss a subtle step or make a mistake.

By automating this process, we can perform various checks upfront, and help the package maintainers to more quickly release a version to NPM in the future.

### PR DESCRIPTION
**Pull request by nvie**
Fixes #118.

This adds a script that performs all the steps from [these instructions](https://www.notion.so/liveblocks/How-to-074d3dfa5b944adb986fe176572d6887#9a10b3d58f014982a513f667ca5d67bd).

This needs to be tested against more edge cases, but it's built in a way where failures of certain steps will not cause harm. The script will end and you can try running it again without issues.

The script is written in a self-checking way, so it will verify its own assumptions before taking action, and fail with a friendly error message otherwise. For example, it won't continue if you have local changes in your worktree, or if you're not up-to-date with the latest upstream version, or if you enter an incorrect version number, etc.

Once I have push access to NPM, I can give this script a spin!

### CODE REVIEW

File: scripts/publish.sh [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+    if [ "$CURRENT_VERSION" != "$VERSION"; then
+        if ! git modified | grep -qEe package-lock.json; then
+            err "Hmm. package-lock.json wasn\'t affected by the version bump. This is fishy. Please manually inspect!"
+            exit 5
+        fi
+    fi
+}
+
+build_pkg () {
+    rm -rf lib
```
Comment (GuillaumeSalles) [inline scripts/publish.sh:234]: We're cleaning `lib` in every `build` scripts so I think we can make the assumptions that every build script should clean every previously built artifacts.

**Comment by GuillaumeSalles** [comment]
That's amazing! Bash is not my cup of tea and I still find your script pretty clear! 

Would it be possible to add a `beta` flag? Could be useful for the first time you're going to publish using this script

**Review by GuillaumeSalles — APPROVED** [review]

File: scripts/publish.sh (hunk 1 above)
Comment (nvie) [reply scripts/publish.sh]: Fantastic. Updated in 1fc1bb442bd49a74748ff617c5f76454c306dc2c.

**Comment by nvie** [comment]
> That's amazing! Bash is not my cup of tea and I still find your script pretty clear!

Glad about that! There is a strict subset of the shell language that I have learned over the years that is safe to use, is platform independent, and avoids bad practices (like not wrapping all uses of variables in quotes, etc.). It doesn't make the scripts more readable, though 🙈 

> Would it be possible to add a `beta` flag? Could be useful for the first time you're going to publish using this script

You mean publish the package under a `beta` tag on NPM? Absolutely, will do! 👍
